waspc/src/Wasp/Project/Common.hs

```diff
@@ -128,3 +131,15 @@ findFileInWaspProjectDir waspDir file = do
   let fileAbsFp = waspDir </> file
   fileExists <- doesFileExist $ toFilePath fileAbsFp
   return $ if fileExists then Just fileAbsFp else Nothing
+
+-- Returns a unique id that can be used for global identification of a specific Wasp project.
+-- Id is no longer than 30 chars, all of them ascii letters, numbers, hyphen or underscore.
+-- It is designed this way to make it easily usable in many scenarios.
+-- It contains app name (or big part of it), to make it also readable for humans.
+-- It is not resistant to Wasp project moving or being renamed, and will change in that case.
+makeAppUniqueId :: Path' Abs (Dir WaspProjectDir) -> String -> String
+makeAppUniqueId waspProjectDir appName = take 19 sanitizedAppName <> "-" <> take 10 projectPathHash
+  where
+    projectPathHash = U.hexToString $ U.checksumFromString $ fromAbsDir waspProjectDir
+    sanitizedAppName = filter isSafeChar appName
+    isSafeChar c = isAsciiLower c || isAsciiUpper c || isDigit c || c == '_' || c == '-'
```

Comment:
How did you decide what is safe, I am curious? This is what works in most places in IT where "identifier" is expected?

If there is indeed a notion of "identifier" that you are thikning about here in the bakcground, tha tmight be interesting to "capture". You could create a small module for that, have `data Identifier`, explain what it stands for, and also have a method from turning string into one and vice versa. That method coudl either do sanitization or fail (maybe those are two methods). Stuff like that. Maybe it is an overkill though. Maybe not hm.

Btw I think Wasp's app name alrady offerrs some guarantees hm. But I guess it is just `String` at the type level? If so yeah probably best to check here again.

Reply:
This is just the moved code (from https://github.com/wasp-lang/wasp/pull/2796#discussion_r2161694117)